### ISSUE TITLE
Cancel Button of Key Modal doesn't work

### ISSUE THREAD

**Issue by Varad25** [mac-s-g/react-json-view#202]
Hello,

Click on cancel button of Key Modal doesn't close the modal.

I can reproduce the issue by starting the dev server on the master branch. 

Am I missing something?

Seems like a problem with v1.18.1

**Comment by franklixuefei** [comment]
@Varad25 Thanks for raising this issue. We recently refactored the codebase so this might be introduced from that. Could you please provide a bit more details on this issue, like steps of reproduction and some screenshots. Thanks!

**Comment by franklixuefei** [comment]
@Varad25 Also include the React version and browser information.

**Comment by Varad25** [comment]
Environment
```
npm -v 5.6.0
node -v v9.8.0
react -v 16.4.1
Browser: Chrome: Version 67.0.3396.99 (Official Build) (64-bit)
```

Steps to reproduce
```
# clone this repository
git clone git@github.com:mac-s-g/react-json-view.git && cd react-json-view
# install dependencies
npm install --save-dev
# run the dev server with hot reloading
npm run dev
```

**Comment by mac-s-g** [comment]
@Varad25 thanks for reporting!  
@franklixuefei i noticed the issue while running through the eslint changes.  I posted a fix with #203 .  would you mind taking a look?
it's from the 16.3 pr.  you fixed some of the less-than-perfect state management, and this particular state change didn't migrate well.
